mailpoet/tests/integration/Router/Endpoints/SubscriptionTest.php

```diff
@@ -18,18 +19,22 @@ class SubscriptionTest extends \MailPoetTest {
   /** @var Captcha */
   private $captcha;
 
+  /*** @var Request */
+  private $request;
+
   public function _before() {
     $this->data = [];
     $this->wp = WPFunctions::get();
     $this->captcha = $this->diContainer->get(Captcha::class);
+    $this->request = $this->diContainer->get(Request::class);
   }
 
   public function testItDisplaysConfirmPage() {
     $pages = Stub::make(Pages::class, [
       'wp' => $this->wp,
       'confirm' => Expected::exactly(1),
     ], $this);
-    $subscription = new Subscription($pages, $this->wp, $this->captcha);
+    $subscription = new Subscription($pages, $this->wp, $this->captcha, $this->request);
```

Comment:
You can simplify this by calling `$this->getServiceWithOverrides()`. It is possible to apply that to each test method.

Reply:
I'm not getting how using this method would make it more simple, at least for these cases.

Reply:
> I'm not getting how using this method would make it more simple, at least for these cases.

If somebody changes dependencies in the future, then he doesn't need to update those lines. But when you think that it's not necessary I'm fine with that.

Reply:
I'd rather leave it like this, I'm more inclined to like that when changing something breaks tests, and the developer has to update the tests, it's a way of keeping informed about the tests, at least for me.